src/EnergyPlus/DataHeatBalance.hh

```diff
@@ -531,20 +542,17 @@ namespace DataHeatBalance {
         EPVector<int> spaces;                           // Pointers to Spaces in the list
     };
 
-    struct ZoneData
+    struct ZoneData : ZoneSpaceData
```

Comment:
Is `ZoneData` really a subclass of `ZoneSpaceData` or is `ZoneSpaceData` a "sub-struct" of `ZoneData`? Is there any usage in which `ZoneSpaceData` can be used instead of `ZoneData`?

Reply:
> Is `ZoneData` really a subclass of `ZoneSpaceData` or is `ZoneSpaceData` a "sub-struct" of `ZoneData`? 

Well, no. `ZoneData` and `SpaceData` are peers with very similar structs (many of the same field names), but then each one has some additional fields that only apply to one or the other.

>Is there any usage in which `ZoneSpaceData` can be used instead of `ZoneData`?

Possibly. For calculating things like Volume, FloorArea, CeilingHeight, etc. I was thinking those could be member functions of `ZoneSpaceData`.

Reply:
Actually, they aren't peers. Spaces are contained within Zones but they do have many similar attributes.

Reply:
Okay, so this is kind of a "tweener" idiom. Not exactly an "is-a" (child struct), not exactly a "has-a" (included struct). Both work for now, can keep it this way until/unless a more obvious parent-child relationship reveals itself.